From 1db79a1e527a09d3f952124265883dffe0e58fc7 Mon Sep 17 00:00:00 2001
From: Roqqit <roqqit@proton.me>
Date: Mon, 17 Jun 2024 17:17:05 +0000
Subject: [PATCH] Lower max block size to 1MB

(cherry picked from commit 3a5f34326d52ae77f1cb7ff2d38c95b486d92fee)

Maintenance rationale (2026-10-14):
Use the Dogecoin consensus block-size limit instead of the much larger default inherited from Bitcoin ABC.

diff --git a/src/config.cpp b/src/config.cpp
index a00ce057517fb29a92fa7b45a69be60f283e4935..0ace8ef3ede7466fbd9749eb3045c0ce45d4945d 100644
--- a/src/config.cpp
+++ b/src/config.cpp
@@ -11,8 +11,7 @@ GlobalConfig::GlobalConfig()
 
 bool GlobalConfig::SetMaxBlockSize(uint64_t maxBlockSize) {
     // Do not allow maxBlockSize to be set below historic 1MB limit
-    // It cannot be equal either because of the "must be big" UAHF rule.
-    if (maxBlockSize <= LEGACY_MAX_BLOCK_SIZE) {
+    if (maxBlockSize < LEGACY_MAX_BLOCK_SIZE) {
         return false;
     }
 
diff --git a/src/consensus/consensus.h b/src/consensus/consensus.h
index 5e7414bb3cdbad5977f635bba612acb7113867cc..4a0b58864bba8476ccf1bea393f4211b1ca5ae29 100644
--- a/src/consensus/consensus.h
+++ b/src/consensus/consensus.h
@@ -21,7 +21,7 @@ static const uint64_t MIN_TX_SIZE = 100;
 /** The maximum allowed size for a block, before the UAHF */
 static const uint64_t LEGACY_MAX_BLOCK_SIZE = ONE_MEGABYTE;
 /** Default setting for maximum allowed size for a block, in bytes */
-static const uint64_t DEFAULT_MAX_BLOCK_SIZE = 32 * ONE_MEGABYTE;
+static const uint64_t DEFAULT_MAX_BLOCK_SIZE = ONE_MEGABYTE;
 /** Allowed number of signature check operations per transaction. */
 static const uint64_t MAX_TX_SIGCHECKS = 3000;
 /**
diff --git a/src/kernel/chainparams.cpp b/src/kernel/chainparams.cpp
index f925491a890992087dd31477cc1afe676e2843d4..032ce2fb71981686226eec88ce53b037346b6f85 100644
--- a/src/kernel/chainparams.cpp
+++ b/src/kernel/chainparams.cpp
@@ -124,7 +124,7 @@ public:
         consensus.gravitonHeight = 0x7ffffffe;
         consensus.phononHeight = 0x7ffffffe;
         consensus.axionHeight = 0x7ffffffe;
-        consensus.wellingtonHeight = 800000; // keep alive for tests
+        consensus.wellingtonHeight = 800000;    // keep alive for tests
         consensus.cowperthwaiteHeight = 900000; // keep alive for tests
         consensus.mengerActivationTime = std::numeric_limits<int64_t>::max();
         consensus.sandersActivationTime = std::numeric_limits<int64_t>::max();
diff --git a/src/policy/policy.h b/src/policy/policy.h
index 253ce2dbb092ff6b1cf408b730b03219ea13140c..780ac19c2773049bc3552fd0b260d90d534d99d6 100644
--- a/src/policy/policy.h
+++ b/src/policy/policy.h
@@ -22,7 +22,7 @@ class CTxOut;
  * Default for -blockmaxsize, which controls the maximum size of block the
  * mining code will create.
  */
-static constexpr uint64_t DEFAULT_MAX_GENERATED_BLOCK_SIZE{16 * ONE_MEGABYTE};
+static constexpr uint64_t DEFAULT_MAX_GENERATED_BLOCK_SIZE{ONE_MEGABYTE - 1000};
 /**
  * Default for -blockmintxfee, which sets the minimum feerate for a transaction
  * in blocks created by mining code.
diff --git a/src/test/config_tests.cpp b/src/test/config_tests.cpp
index 50ed059e0aa835681a4f4de7d2e3dba32c207287..c8c3ec896402aa77a52ae8cc79023dcb2b4e61bf 100644
--- a/src/test/config_tests.cpp
+++ b/src/test/config_tests.cpp
@@ -21,9 +21,9 @@ BOOST_AUTO_TEST_CASE(max_block_size) {
     BOOST_CHECK(!config.SetMaxBlockSize(0));
     BOOST_CHECK(!config.SetMaxBlockSize(12345));
     BOOST_CHECK(!config.SetMaxBlockSize(LEGACY_MAX_BLOCK_SIZE - 1));
-    BOOST_CHECK(!config.SetMaxBlockSize(LEGACY_MAX_BLOCK_SIZE));
 
-    // LEGACY_MAX_BLOCK_SIZE + 1
+    // Smallest
+    BOOST_CHECK(config.SetMaxBlockSize(LEGACY_MAX_BLOCK_SIZE));
     BOOST_CHECK(config.SetMaxBlockSize(LEGACY_MAX_BLOCK_SIZE + 1));
     BOOST_CHECK_EQUAL(config.GetMaxBlockSize(), LEGACY_MAX_BLOCK_SIZE + 1);
 
diff --git a/src/test/util/setup_common.cpp b/src/test/util/setup_common.cpp
index 96da8098506a46cea17530ff8d753e1ac8294cde..359f7fd173af751f8ab3e2accb45c1ddab62fb29 100644
--- a/src/test/util/setup_common.cpp
+++ b/src/test/util/setup_common.cpp
@@ -315,7 +315,7 @@ TestChain100Setup::TestChain100Setup(
         LOCK(::cs_main);
         assert(
             m_node.chainman->ActiveTip()->GetBlockHash().ToString() ==
-            "8916b4cf6a6dbb901536365ed5541e39f9ec3b95880762300d0401360bd9cce9");
+            "349766ba691ce76e0d2a0b27656004913a2e363510be135f6d660d50197aa7ce");
     }
 }
 
diff --git a/test/functional/abc_p2p_compactblocks.py b/test/functional/abc_p2p_compactblocks.py
index f01916f9f249e78d85b118aba1df7caad462d794..3bae0247ce09266e482f08ac7f5f9bda698368e6 100644
--- a/test/functional/abc_p2p_compactblocks.py
+++ b/test/functional/abc_p2p_compactblocks.py
@@ -277,7 +277,7 @@ class FullBlockTest(BitcoinTestFramework):
         test_p2p.send_without_ping(test_p2p.last_headers)
 
         # Send a block
-        b1 = block(1, spend=out[0], block_size=ONE_MEGABYTE + 1)
+        b1 = block(1, spend=out[0], block_size=int(ONE_MEGABYTE / 2))
         default_p2p.send_blocks_and_test([self.tip], node)
 
         # Checks the node to forward it via compact block
@@ -295,7 +295,7 @@ class FullBlockTest(BitcoinTestFramework):
         b2 = block(
             2,
             spend=out[1],
-            extra_txns=70000,
+            extra_txns=9000,
             block_size=self.excessive_block_size - 1000,
         )
         default_p2p.send_blocks_and_test([self.tip], node)
